source/userguide/dm.rst

```diff
@@ -659,6 +658,102 @@ For the use case of transferring money in a banking assistant application, the f
                    "{account_from} to a {account_to}. Is that right?"]
         responder.reply(replies)
 
+
+Subset forms and invoking slot-filling within in a dialogue handler
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To support dynamically iterating through a form, we suggest the use of subset forms. Consider a case in which the initial form fetches a certain set of entities, and the flow ahead depends on the values of those captured entities. This requires capturing of further entities based on those values. For this, we recommend using subset forms.
+
+Take into consideration the previous example. Now, instead of a single pass form to capture all entities, say we want to check whether the account from which money is withdrawn is a savings account or not. If yes, then continue to fetch the other entities (using ``form_transfermoney_2``), otherwise send appropriate response.
+
+.. code:: python
+
+    from mindmeld.components.dialogue import AutoEntityFilling
+    from mindmeld.core import FormEntity
+
+    form_transfermoney_1 = {
+        'entities':[
+            FormEntity(
+                entity='account_type',
+                role='account_from',
+                responses=['Sure. Transfer from which account?']
+                ),
+            ]
```

Comment:
There are no exit keys and exit msg in form 1?

Reply:
That's an optional param.

Reply:
Probably for completeness we should add them to avoid confusion on the side of the developer